### ISSUE TITLE
Require 2 python versions?

### ISSUE THREAD

**Issue by leo-liuzy** [xlang-ai/text2reward#1]
I am setting up the environment. 

What I found is that in the setup tutorial, you have 
`conda create -n text2reward python=3.7`

However, langchain which is used in `text2reward/code_generation/single_flow/*/generation.py` requires 3.8. 

# installing with 3.7
If I am installing langchain with 3.7. It will not install `langchain.chat_models` and `langchain.callbacks.manager` and making the code-generation no runnable.

# installing with 3.8
This will make the following part not runnable (Using python3.7 doesn't have this issue)
```
cd ManiSkill2
pip install -e .
pip install stable-baselines3==1.8.0 wandb tensorboard
```
For the above install, I will get 
```
Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [1 lines of output]
      error in gym setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.
```
I try to fix `error in gym setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.`. But following [here](https://github.com/giampaolo/psutil/issues/1268) doesn't make things working.

**Comment by Timothyxxx** [comment]
You are right, I tried to reproduce and found this issue as well. It's a conflict caused by `mujoco` (which requires a lower version of python) and `langchain` (which requires a higher version of python), we will update and fix this error.

**Comment by songyuc** [comment]
Hi,
I'm also encountering this issue. When using Python 3.7, some required modules (like `langchain.chat_models` and `langchain.callbacks.manager`) are missing, causing the code generation part to fail. On the other hand, upgrading to Python 3.8 resolves the `langchain` dependency issues but then leads to errors during installing the ManiSkill2 environment. 

Is there any update or workaround available to address this version conflict between langchain and the other dependencies (such as `mujoco`)? Any guidance would be greatly appreciated.

Thanks!

**Comment by sihengz02** [comment]
I believe in 99% scenarios you do not need to generate the code and run the ManiSkill2 environment at the same time. So one workaround may be that you can use two different conda envs for the code generation part and the RL part.
